gateway/testutil.go

```diff
@@ -1030,6 +1030,14 @@ func (s *Test) newGateway(genConf func(globalConf *config.Config)) *Gateway {
 	gw := NewGateway(gwConfig, s.ctx)
 	gw.setTestMode(true)
 
+	// This overrides the default dialer for redis to always use 127.0.0.1 in tests;
+	// It also avoids dns lookups, the dns mock.
+	gw.RedisController.SetConstructor(func(cache bool, analytics bool, config config.Config) redis.UniversalClient {
+		return storage.NewRedisClusterPoolWithOptions(cache, analytics, config, func(cfg *redis.UniversalOptions) {
+			cfg.Dialer = test.LocalDialer()
```

Comment:
So this is the actual race fix?

Reply:
No, the `-StartTest` in testPrepareHMACAuthSessionPass(); the local dialer only dials `127.0.0.1`; gateway by default doesn't even pass the tests on macs because `localhost` is ipv6 or something and doesn't work. I guess this fixes tests on macs, or maybe just jeffy's machine.